@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -710,28 +714,13 @@ function placePointLabelAtAnchor(
     tmpSize.multiplyScalar(scale);
     tmp2DBox.set(tmpCenter.x - tmpSize.x / 2, tmpCenter.y - tmpSize.y / 2, tmpSize.x, tmpSize.y);
 
-    // Update output screen position.
-    outScreenPosition.set(textOffset.x, textOffset.y, labelState.renderDistance);
-
-    // Check the text visibility if invisible finish immediately
-    // regardless of the persistence state - no fading required.
-    if (!screenCollisions.isVisible(tmp2DBox)) {
-        return PlacementResult.Invisible;
-    }
```

Comment:
I don't understand why you postpone this check. Why do you wan't to check collisions if the text is invisible?

Reply:
> If we declare all text parts of the poi labels "invisible", in the current flow of the code, the icons are not rendered as well. 

With your changes that's not true anymore:
https://github.com/heremaps/harp.gl/pull/1888/files#diff-6b6f266bec84f376c6f75070ad562145d6e14b8e4031e0e34c85be854c939f39R1664-R1667 

> it makes sense to reserve space outside the current visible space

We aren't reserving any space for invisible text later in the function:
https://github.com/heremaps/harp.gl/pull/1888/files#diff-cbaa0c98ceb02c4586a463f1982b813f5555d33eab176cd4f180be8eaad25e6cR742

So the only difference I see is that we would mark colliding labels outside of the screen as rejected instead of invisible.
It'd be good to know exactly why this behaviour is better, cause we do it at the expense of checking more collisions (a perf bottleneck).

Reply:
re-added the test

Reply:
This condition is not needed anymore then:
https://github.com/heremaps/harp.gl/pull/1888/files#diff-cbaa0c98ceb02c4586a463f1982b813f5555d33eab176cd4f180be8eaad25e6cR738

It's the same test you added back.